core/src/parser/mod.rs

```diff
@@ -155,8 +155,44 @@ impl<'ast> Parser<'ast> {
     }
 
     #[inline]
+    fn param_from_expression(&mut self, expression: Expression<'ast>) -> Parameter<'ast> {
+        match expression {
+            Expression::Identifier(ident) => {
+                Parameter {
+                    key: ParameterKey::Identifier(ident),
+                    value: None
+                }
+            },
+            Expression::Binary {
+                operator: OperatorKind::Assign,
+                left,
+                right
+            } => {
+                Parameter {
+                    key: self.param_from_expression(left.item).key,
+                    value: Some(right)
+                }
+            },
+            _ => {
+                panic!("Unexpected token")
```

Comment:
More of a TODO, but this should be an error of some kind instead of a panic (could just push to the errors Vec and return an empty `Parameter`).